### DISCUSSION TITLE
Add a way to fine tune `otDatasetCreateNewNetwork` results

### DISCUSSION THREAD

**Discussion by rretanubun** [openthread/openthread#9732] [Ideas]
If I understands things correctly, having the ability to pre-set the channel mask before calling `otDatasetCreateNewNetwork`
which maps to :

https://github.com/openthread/openthread/blob/3400692b03044b9c673d3e015c24f9c0cd7cb404/src/core/meshcop/dataset.cpp#L54-L77

will allow something like this (paraphrased using some missing APIs):

```
otPlatRadioSetSupportedChannelMask(aNonDefaultChannelMask); // For example, limit channels to 11-to-25 to avoid 26
otDatasetCreateNewNetwork(); // The newly formed dataset is influence by the previous call and pick a random channel between 11-to-25
```

Or is there an existing or alternate method to achieve the same thing that I am missing?

Thanks for everyone's time!

**Comment by abtink** [comment]
@rretanubun what you say is correct.

Also want to point out that `otDatasetCreateNewNetwork()` will create a new dataset and return it in the given `otOperationalDataset *aDataset. Caller can then change any fields in `otOperationalDataset` object itself afterwards.

**Reply by rretanubun** [reply]
@abtink : Yep, I am doing what you suggested to achieve what I need for now, just wanted to optimize away the `extra work` with the `idea`. Thanks for confirming my understanding.
